src/NServiceBus.Transport.RabbitMQ.Tests/RabbitMqContext.cs

```diff
@@ -13,53 +17,61 @@ class RabbitMqContext
         public virtual int MaximumConcurrency => 1;
 
         [SetUp]
-        public void SetUp()
+        public async Task SetUp()
         {
-            routingTopology = new ConventionalRoutingTopology(true);
             receivedMessages = new BlockingCollection<IncomingMessage>();
 
             var connectionString = Environment.GetEnvironmentVariable("RabbitMQTransport_ConnectionString");
 
             if (string.IsNullOrEmpty(connectionString))
             {
-                throw new Exception("The 'RabbitMQTransport_ConnectionString' environment variable is not set.");
+                connectionString = "host=localhost";
+                //throw new Exception("The 'RabbitMQTransport_ConnectionString' environment variable is not set.");
             }
 
-            var config = ConnectionConfiguration.Create(connectionString, ReceiverQueue);
-
-            connectionFactory = new ConnectionFactory(ReceiverQueue, config,  default, false, false, default, default);
-            channelProvider = new ChannelProvider(connectionFactory, config.RetryDelay, routingTopology);
-            channelProvider.CreateConnection();
-
-            messageDispatcher = new MessageDispatcher(channelProvider);
-
-            var purger = new QueuePurger(connectionFactory);
-
-            messagePump = new MessagePump(connectionFactory, new MessageConverter(), "Unit test", channelProvider, purger, TimeSpan.FromMinutes(2), 3, 0);
+            var connectionStringBuilder = new DbConnectionStringBuilder { ConnectionString = connectionString };
 
-            routingTopology.Reset(connectionFactory, new[] { ReceiverQueue }.Concat(AdditionalReceiverQueues), new[] { ErrorQueue });
+            //TODO: Parse any settings in the connection string 
+            var transport = new RabbitMQTransport { Host = (string)connectionStringBuilder["Host"] };
```

Comment:
Should the connetion string be obligatory constructor paramter?

Reply:
I don't think so. The only obligatory thing that needs to be set is `Host`. Hard to decide if `RabbitMQTransport(string)` ctor should take a connection string or the host.